src/main/java/org/apache/sysds/runtime/privacy/propagation/PrivacyPropagator.java

```diff
@@ -64,7 +64,61 @@ public static Data parseAndSetPrivacyConstraint(Data cd, JSONObject mtd)
 		}
 		return cd;
 	}
-	
+
+	private static boolean anyInputHasLevel(PrivacyLevel[] inputLevels, PrivacyLevel targetLevel){
+		return Arrays.stream(inputLevels).anyMatch(i -> i == targetLevel);
+	}
+
+	/**
+	 * Returns the output privacy level based on the given input privacy levels and operator type.
+	 * It represents the basic logic of privacy propagation:
+	 *
+	 * Unary input:
+	 * Input   | NonAggregate | Aggregate
+	 * -----------------------------------
+	 * priv    | priv         | priv
+	 * privAgg | privAgg      | none
+	 * none    | none         | none
+	 *
+	 * Binary input:
+	 * Input   			| NonAggregate 	| Aggregate
+	 * -----------------------------------
```

Comment:
minor alignment

Reply:
I am not sure, what you believe should be aligned, but now I added additional "-------".